pkg/operator/ceph/object/spec.go

```diff
@@ -235,8 +261,18 @@ func (c *clusterConfig) createCaBundleUpdateInitContainer(rgwConfig *rgwConfig)
 // are copied to other volume from mounted secrets then ownership/permissions are changed
 // accordingly with help of an init container.
 func (c *clusterConfig) vaultTokenInitContainer(rgwConfig *rgwConfig) v1.Container {
-	_, srcVaultVolMount := kms.VaultVolumeAndMount(c.store.Spec.Security.KeyManagementService.ConnectionDetails, "")
+	var vaultVolMounts []v1.VolumeMount
+
 	tmpVaultMount := v1.VolumeMount{Name: rgwVaultVolumeName, MountPath: rgwVaultDirName}
+	vaultVolMounts = append(vaultVolMounts, tmpVaultMount)
+	if kmsEnabled, _ := c.CheckRGWKMS(); kmsEnabled {
+		_, ssekmsVaultVolMount := kms.VaultVolumeAndMountWithCustomName(c.store.Spec.Security.KeyManagementService.ConnectionDetails, "", sseKMS)
+		vaultVolMounts = append(vaultVolMounts, ssekmsVaultVolMount)
+	}
+	if s3Enabled, _ := c.CheckRGWSSES3Enabled(); s3Enabled {
+		_, sses3VaultVolMount := kms.VaultVolumeAndMountWithCustomName(c.store.Spec.Security.ServerSideEncryptionS3.ConnectionDetails, "", sseS3)
+		vaultVolMounts = append(vaultVolMounts, sses3VaultVolMount)
+	}
```

Comment:
These instances of swallowing errors haven't been fixed. If we already know kms/s3 enabled status, we should pass that to this function. If not, we should return the error from this function.